src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs

```diff
@@ -0,0 +1,100 @@
+        {
+        }
+
+        [Theory()]
+        [MemberData(nameof(SupportedTfms))]
+        public void TrimmingOptions_Are_Defaulted_Correctly_On_Trimmed_Apps(string targetFramework)
+        {
+            var projectName = "HelloWorld";
+            var rid = EnvironmentInfo.GetCompatibleRid(targetFramework);
+
+            var testProject = CreateTestProjectForILLinkTesting(targetFramework, projectName);
+            testProject.AdditionalProperties["PublishTrimmed"] = "true";
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: projectName + targetFramework);
+
+            var publishCommand = new PublishCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+            publishCommand.Execute($"/p:RuntimeIdentifier={rid}").Should().Pass();
+
+            string outputDirectory = publishCommand.GetOutputDirectory(targetFramework: targetFramework, runtimeIdentifier: rid).FullName;
+            string runtimeConfigFile = Path.Combine(outputDirectory, $"{projectName}.runtimeconfig.json");
+            string runtimeConfigContents = File.ReadAllText(runtimeConfigFile);
+
+            JsonNode runtimeConfig = JsonNode.Parse(runtimeConfigContents);
+            JsonNode configProperties = runtimeConfig["runtimeOptions"]["configProperties"];
+
+            configProperties["Microsoft.AspNetCore.EnsureJsonTrimmability"].GetValue<bool>()
+                    .Should().BeTrue();
+        }
+
+        [Theory()]
+        [MemberData(nameof(SupportedTfms))]
+        public void TrimmingOptions_Are_Defaulted_Correctly_On_Aot_Apps(string targetFramework)
+        {
+            var projectName = "HelloWorld";
+            var rid = EnvironmentInfo.GetCompatibleRid(targetFramework);
+
+            var testProject = CreateTestProjectForILLinkTesting(targetFramework, projectName);
+            testProject.AdditionalProperties["PublishAOT"] = "true";
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: projectName + targetFramework);
+            var publishCommand = new PublishCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+            publishCommand.Execute($"/p:RuntimeIdentifier={rid}").Should().Pass();
+
+            string outputDirectory = publishCommand.GetIntermediateDirectory(targetFramework: targetFramework, runtimeIdentifier: rid).FullName;
+            string responseFile = Path.Combine(outputDirectory, "native", $"{projectName}.ilc.rsp");
+            var responseFileContents = File.ReadLines(responseFile);
+
+            responseFileContents.Should().Contain("--feature:Microsoft.AspNetCore.EnsureJsonTrimmability=true");
+        }
+
+        public static IEnumerable<object[]> SupportedTfms { get; } = new List<object[]>
+        {
+#if NET8_0
+            new object[] { ToolsetInfo.CurrentTargetFramework }
+#else
+#error If building for a newer TFM, please update the values above
+#endif
```

Comment:
Is this the typical way we add a `TODO` when a new TFM gets added to add support for previous TFMs?

Reply:
I borrow this from https://github.com/dotnet/sdk/blob/7a05724a3a1aaf0c0f97f098ae174e3faf2589aa/src/Tests/Microsoft.NET.Publish.Tests/PublishTestUtils.cs#L10